apps/teams-test-app/src/components/DialogAPIs.tsx

```diff
@@ -16,29 +16,56 @@ const DialogAPIs = (): ReactElement => {
             throw new Error('Url undefined');
           }
         },
-        submit: async (dialogInfo, setResult) => {
-          const onComplete = (err: string, result: string | object): void => {
-            setResult('Error: ' + err + '\nResult: ' + result);
-          };
-          // Store the reference of child window in React
-          const childWindow = dialog.open(dialogInfo, onComplete);
-          childWindow.addEventListener('message', (message: string) => {
-            // Message from parent
-            setResult(message);
-          });
-          childWindowRef.current = childWindow;
-          return '';
+        submit: {
+          withPromise: async (dialogInfo, setResult) => {
+            const onComplete = (err: string, result: string | object): void => {
+              setResult('Error: ' + err + '\nResult: ' + result);
+            };
+            // Store the reference of child window in React
+            const childWindow = dialog.open(dialogInfo, onComplete);
+            childWindow.addEventListener('message', (message: string) => {
+              // Message from parent
+              setResult(message);
+            });
+            childWindowRef.current = childWindow;
```

Comment:
can this be put into some sort of a helper function to avoid duplication?

Reply:
Can you use the helper for promise version too?

Reply:
Yes. I am still doing the test to check the e2e test failure. I should put that in the PR description. Sorry about that.